etna/feedback/models.py

```diff
@@ -0,0 +1,262 @@
+        else:
+            # If the prompt has been defined with a trailing slash, but
+            # this request does not have one, do not allow a match
+            exact_path_variations = (path,)
+
+        obj = (
+            self.annotate(
+                match=Case(
+                    When(
+                        path__in=exact_path_variations,
+                        startswith_path=False,
+                        then=constants.MATCH_EXACT_PATH,
+                    ),
+                    When(
+                        path=exact_path_variations,
+                        startswith_path=True,
+                        then=constants.MATCH_EXACT_PATH_WITH_SUB_PATHS,
+                    ),
+                    default=constants.MATCH_SUB_PATH,
+                    output_field=IntegerField(),
+                )
+            )
+            .filter(
+                Q(path__in=exact_path_variations)
+                | Q(path__in=presceeding_paths, startswith_path=True)
+            )
+            .order_by("match", Length("path").desc())
+            .first()
+        )
+        if obj is None:
+            raise FeedbackPrompt.DoesNotExist
+        return obj
+
+    def get_by_natural_key(self, public_id: Union[str, uuid.UUID]) -> "FeedbackPrompt":
+        return self.get(public_id=public_id)
+
+
+@register_snippet
+class FeedbackPrompt(RevisionMixin, ClusterableModel):
+    public_id = models.UUIDField(
+        editable=False, unique=True, default=uuid.uuid4, verbose_name=_("public ID")
+    )
+    text = models.CharField(
+        verbose_name=_("prompt text"),
+        max_length=constants.PROMPT_TEXT_MAX_LENGTH,
+        default="What did you think of this page?",
+    )
+    response_options = StreamField(
+        [("option", ResponseOptionBlock())],
+        use_json_field=True,
+        verbose_name=_("response options"),
+    )
+    thank_you_heading = models.CharField(
+        max_length=100,
+        verbose_name=_("thank you heading"),
+        default=_("Thank you for your valuable feedback"),
+        help_text=_("Displayed to users after succesfully submitting feedback."),
+    )
+    thank_you_message = RichTextField(
+        max_length=200,
+        verbose_name=_("thank you message"),
+        features=settings.RESTRICTED_RICH_TEXT_FEATURES,
+        help_text=_(
+            "If supplied, displayed below the 'Thank you heading' when a user succesfully submits feedback."
+        ),
+        blank=True,
+    )
+    continue_link_text = models.CharField(
+        max_length=200,
+        default=_("Return to the previous page"),
+        help_text=_(
+            'After submitting feedback, non-JS users are taken to a success page, where the thank you headding and text are displayed, followed by a link to return to the previous page. This field can be used to change the text of that link. For example, "Continue with your search" might make more sense to users.'
+        ),
+    )
```

Comment:
First of all, you will need to disable JS, so that the form posts in the normal way (not via Ajax). Alternatively, you can comment out the `<script>` tag from within the `prompt.html` template.

This field is used as link text for the "Continue with what you were doing" link that appears on the success page... which is irrelevant when JS is enabled, because you don't actually leave the page.